crates/risc0-bonsai/src/host.rs

```diff
@@ -293,30 +295,62 @@ impl<'a> ZkvmHost for Risc0BonsaiHost<'a> {
         Risc0Guest::with_hints(std::mem::take(&mut self.env))
     }
 
+    fn add_assumption(&mut self, receipt_buf: Vec<u8>) {
+        let receipt: Receipt = bincode::deserialize(&receipt_buf).expect("Receipt should be valid");
+        let ass: AssumptionReceipt = receipt.into();
+        let a = ass.clone().claim_digest().unwrap();
+        println!("Digest of assumption: {:?}", a);
+
+        self.assumptions.push(ass);
+    }
+
     /// Only with_proof = true is supported.
     /// Proofs are created on the Bonsai API.
     fn run(&mut self, with_proof: bool) -> Result<Proof, anyhow::Error> {
         let proof = match (self.client.as_ref(), with_proof) {
             // Local execution. If mode is Execute, we always do local execution.
             (_, false) => {
-                let env = add_benchmarking_callbacks(ExecutorEnvBuilder::default())
-                    .write_slice(&self.env)
-                    .build()
-                    .unwrap();
-                let mut executor = ExecutorImpl::from_elf(env, self.elf)?;
+                // Set dev mode to true as this is used for tests
+                std::env::set_var("RISC0_DEV_MODE", "1");
+                let mut env = add_benchmarking_callbacks(ExecutorEnvBuilder::default());
+                for assumption in self.assumptions.iter() {
+                    env.add_assumption(assumption.clone());
+                    tracing::info!("added assumption to the env");
+                }
+
+                let env = env.write_slice(&self.env).build().unwrap();
+
+                let prover = LocalProver::new("citrea-test-prover");
+
+                tracing::info!("Starting risc0 proving");
+                let ProveInfo { receipt, stats } = prover.prove(env, self.elf)?;
+
+                println!(
+                    "\n\n\nreceipt journal after proving: {:?} \n\n\n",
```

Comment:
Is this a left over?
```suggestion
                    "receipt journal after proving: {:?}",
```

Or should the whole print be removed?

Reply:
Removed